site/documentation/content/api/telemetry-pubsub/index.md

```diff
@@ -0,0 +1,118 @@
+| *orderingKey* | *string*      | If non-empty, identifies related messages for which publish order should be respected. If a Subscription has enableMessageOrdering set to true, messages published with the same non-empty orderingKey value will be delivered to subscribers in the order in which they are received by the Pub/Sub system. |
+
+Metadata MUST be set as attributes on a Pub/Sub message.
+The following table provides an overview a client needs to set on a *Publish Telemetry Data* message.
+
+| Name            | Mandatory       | Type        | Description |
+| :-------------- | :-------------: | :---------- | :---------- |
+| *content-type*  | yes             | *string*    | A content type indicating the type and characteristics of the data contained in the message value as a valid MIME type, e.g. `text/plain; charset="utf-8"` for a text message or `application/json` etc. The value may be set to `application/octet-stream` if the message message value is to be considered *opaque* binary data. See [RFC 2046](https://www.ietf.org/rfc/rfc2046.txt) for details. |
+| *content-encoding* | no           | *string*    | The content encoding as defined in section 3.5 of [RFC 2616](https://www.ietf.org/rfc/rfc2616.txt). |
+| *creation-time* | yes             | *long*      | The instant in time (UTC, milliseconds since the Unix epoch) when the message has been created. |
+| *device_id*     | yes             | *string*    | The identifier of the device that the data in the message value is originating from. |
+| *qos*           | no              | *int*       | The quality of service level of the message. Supported values are `0` for AT MOST ONCE and `1` for AT LEAST ONCE. |
+| *ttd*           | no              | *int*       | The *time 'til disconnect* indicates the number of seconds that the device will remain connected to the protocol adapter. The value of this header must be interpreted relative to the message's *creation-time*. A value of `-1` is used to indicate that the device will remain connected until further notice, i.e. until another message indicates a *ttd* value of `0`. In absence of this property, the connection status of the device is to be considered indeterminate. *Backend Applications* might use this information to determine a time window during which the device will be able to receive a command. |
+| *ttl*           | no              | *long*      | The *time-to-live* in milliseconds. A consumer of the message SHOULD discard the message if the sum of *creation-time* and *ttl* is greater than the current time (milliseconds since the Unix epoch). |
+
+Protocol adapters MAY set additional attributes on the Pub/Sub message.
+
+The value of the message MUST consist of a byte array containing the telemetry data. The format and encoding of the
+data MUST be indicated by the *content-type* and (optional) *content-encoding* attributes of the message.
```

Comment:
This does not seem to match with the indicated `string' type of the *data* field?